### PR TITLE
[DEV-1970] add fee paymentMethod

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
Add migration

File: src/subdomains/core/buy-crypto/routes/crypto-route/crypto-route.controller.ts [hunk 1]
```diff
@@ -154,7 +157,13 @@ export class CryptoRouteController {
...
     );
 
-    const fee = await this.userService.getUserFee(userId, FeeDirectionType.CONVERT, crypto.asset, minFee.amount);
+    const fee = await this.userService.getUserFee(
+      userId,
+      CryptoPaymentMethod.CRYPTO,
+      FiatPaymentMethod.BANK,
+      crypto.asset,
+      minFee.amount,
+    );
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/crypto-route/crypto-route.controller.ts:166]: Crypto -> Crypto?

File: src/subdomains/supporting/payment/dto/create-fee.dto.ts [hunk 2]
```diff
@@ -44,8 +44,14 @@ export class CreateFeeDto {
...
-  @IsEnum(FeeDirectionType)
-  direction: FeeDirectionType;
+  @IsEnum([FiatPaymentMethod, CryptoPaymentMethod])
+  @IsArray()
+  paymentMethodIns: (FiatPaymentMethod | CryptoPaymentMethod)[];
+
+  @IsOptional()
+  @IsEnum([FiatPaymentMethod, CryptoPaymentMethod])
+  @IsArray()
+  paymentMethodOuts: (FiatPaymentMethod | CryptoPaymentMethod)[];
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/dto/create-fee.dto.ts:54]: `paymentMethodsIn` & `paymentMethodsOut`?

File: src/subdomains/supporting/payment/entities/fee.entity.ts [hunk 3]
```diff
@@ -44,8 +42,11 @@ export class Fee extends IEntity {
   @Column({ length: 256, nullable: true })
   accountType: AccountType;
 
-  @Column({ length: 256, nullable: true })
-  direction: FeeDirectionType;
+  @Column({ length: 'MAX', nullable: true })
+  paymentMethodIn: string;
+
+  @Column({ length: 'MAX', nullable: true })
+  paymentMethodOut: string;
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/entities/fee.entity.ts:49]: Method**s**?

File: src/subdomains/supporting/payment/services/transaction-helper.ts [hunk 4]
```diff
@@ -346,9 +366,9 @@ export class TransactionHelper implements OnModuleInit {
...
 
-  private getTxInfo(from: Asset | Fiat, to: Asset | Fiat): { direction: FeeDirectionType; feeAsset: Asset } {
-    if (from instanceof Fiat && to instanceof Asset) return { direction: FeeDirectionType.BUY, feeAsset: to };
-    if (from instanceof Asset && to instanceof Fiat) return { direction: FeeDirectionType.SELL, feeAsset: from };
-    return { direction: FeeDirectionType.CONVERT, feeAsset: to instanceof Asset ? to : undefined };
+  private getTxInfo(from: Asset | Fiat, to: Asset | Fiat): Asset {
+    if (from instanceof Fiat && to instanceof Asset) return to;
+    if (from instanceof Asset && to instanceof Fiat) return from;
+    return to instanceof Asset ? to : undefined;
   }
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/services/transaction-helper.ts:373]: Rename to `getFeeAsset`?

**Review by davidleomay — APPROVED** [review]

**Comment by Yannick1712** [comment]
deactivate preparation services before merging, update fee paymentMethods

**Pull request merged**